addons/payment_ogone/controllers/main.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import pprint
+
+import werkzeug
+
+from odoo import _, http
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class OgoneController(http.Controller):
+    _accept_url = '/payment/ogone/accept'
+    _decline_url = '/payment/ogone/ogonedecline'
+    _exception_url = '/payment/ogone/exception'
+    _cancel_url = '/payment/ogone/cancel'
+    _fleckcheckout_url = '/payment/ogone/flexchekout/feedback'
+    _fleckcheckout_final_url = "/payment/ogone/flexchekout/final"
+
+    @http.route('/payment/ogone/payment_setup', type='json', auth='public')
+    def payment_setup(self, acquirer_id, amount=None, currency_id=None, partner_id=None, **data):
+        """ Setup the Ogone iframe used with the FlexCheckout API.
```

Comment:
```suggestion
    @http.route('/payment/ogone/build_iframe_url', type='json', auth='public')
    def ogone_build_iframe_url(self, acquirer_id, amount=None, currency_id=None, partner_id=None, **data):
        """ Build the URL of the iframe of the FlexCheckout API with all necessary parameters.
```
This suggestion looks nitpicking but I find that it adds a non-negligible amount of clarity to the flow of Ogone which is complex enough on its own.
Since you already have the acquirer_id in the client, this route can very well be used to (only) build the iframe URL and return it.